.travis.yml

```diff
@@ -0,0 +1,33 @@
+# the following values must be set in https://travis-ci.com/<github-repo>/settings
+# - DOCKER_REPO (eg. myrepo/myapp)
+# - DOCKER_USERNAME
+# - DOCKER_PASSWORD
+
+services: docker
+language: go
+
+branches:
+  only:
+    - master
+    - /^v\d+\.\d+\.\d+.*$/
+
+env:
+  - ARCH=amd64
+  - ARCH=arm32v7
+  - ARCH=arm64v8
+
+before_script:
+  - docker run --rm --privileged multiarch/qemu-user-static:register --reset
+
+script:
+  - make build ARCH=${ARCH} DOCKER_REPO=${DOCKER_REPO}
+  - make test ARCH=${ARCH} DOCKER_REPO=${DOCKER_REPO}
+
+after_success:
+  # uncomment the following line to deploy only when a new git tag is pushed
+  # otherwise it will deploy on every push
+  # - test ${TRAVIS_TAG} =~ /^v\d+\.\d+\.\d+.*$/ || exit 0
```

Comment:
Is there any expected behaviour or best practice to have a deployment for every commit to master branch? Or indeed every commit on every branch? In other words, why would this ever need to be *commented*?

Reply:
I only had it commented for testing my build hooks (I didn't want to tag it every time), but I think you'd always want to filter deployment by tag or by branch, depending on your build process.
Maybe uncomment it once you've confirmed that your travis-ci configuration is working? I like having the build portion (no deploy) run on every commit as sort of a sanity, but that's up to you as well.